src/openrct2/ride/Vehicle.cpp

```diff
@@ -4830,6 +4830,24 @@ void Vehicle::UpdateSimulatorOperating()
     var_C0 = 0;
 }
 
+void UpdateRotatingDefault(Vehicle& vehicle)
+{
+    vehicle.sub_state = 1;
+    vehicle.UpdateRotating();
+}
+
+void UpdateRotatingEnterprise(Vehicle& vehicle)
+{
+    if (vehicle.sub_state == 2)
+    {
+        vehicle.SetState(Vehicle::Status::Arriving);
+        vehicle.var_C0 = 0;
+        return;
+    }
+    vehicle.sub_state = 1;
```

Comment:
Would it perhaps make sense to call `UpdateRotatingDefault(vehicle)` here instead of duplicating it. I'm not entirely sure myself if that makes sense.